test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java

```diff
@@ -0,0 +1,161 @@
+        public static final Version V_5_3_1 = Version.fromString("5.3.1");
+        public static final Version V_5_3_2 = Version.fromString("5.3.2");
+        public static final Version V_5_4_0 = Version.fromString("5.4.0");
+        public static final Version V_5_4_1 = Version.fromString("5.4.1");
+        public static final Version CURRENT = V_5_4_1;
+    }
+    public void testResolveReleasedVersionsForReleaseBranch() {
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestReleaseBranch.CURRENT, TestReleaseBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(Arrays.asList(TestReleaseBranch.V_5_3_0, TestReleaseBranch.V_5_3_1, TestReleaseBranch.V_5_3_2,
+                TestReleaseBranch.V_5_4_0, TestReleaseBranch.V_5_4_1), released);
+        assertEquals(emptyList(), unreleased);
+    }
+
+    static class TestStableBranch {
+        public static final Version V_5_3_0 = Version.fromString("5.3.0");
+        public static final Version V_5_3_1 = Version.fromString("5.3.1");
+        public static final Version V_5_3_2 = Version.fromString("5.3.2");
+        public static final Version V_5_4_0 = Version.fromString("5.4.0");
+        public static final Version CURRENT = V_5_4_0;
+    }
+    public void testResolveReleasedVersionsForUnreleasedStableBranch() {
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestStableBranch.CURRENT,
+                TestStableBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(
+                Arrays.asList(TestStableBranch.V_5_3_0, TestStableBranch.V_5_3_1, TestStableBranch.V_5_4_0),
+                released);
+        assertEquals(singletonList(TestStableBranch.V_5_3_2), unreleased);
+    }
+
+    static class TestUnstableBranch {
+        public static final Version V_5_3_0 = Version.fromString("5.3.0");
+        public static final Version V_5_3_1 = Version.fromString("5.3.1");
+        public static final Version V_5_3_2 = Version.fromString("5.3.2");
+        public static final Version V_5_4_0 = Version.fromString("5.4.0");
+        public static final Version V_5_5_0 = Version.fromString("5.5.0");
+        public static final Version CURRENT = V_5_5_0;
+    }
+    public void testResolveReleasedVersionsForUnstableBranch() {
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestUnstableBranch.CURRENT,
+                TestUnstableBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(Arrays.asList(TestUnstableBranch.V_5_3_0, TestUnstableBranch.V_5_3_1, TestUnstableBranch.V_5_5_0), released);
+        assertEquals(Arrays.asList(TestUnstableBranch.V_5_3_2, Version.V_5_4_0), unreleased);
+    }
+
+    static class TestUnstableBranchWithAlphas {
```

Comment:
Do you think I should drop the test without alphas then? Or rename it? I'd kind of like to keep it because it is telling is only one of the two tests fails.